agent/lib/kontena/rpc_client.rb

```diff
@@ -52,21 +53,28 @@ def notification(method, params)
     # @raise abort
     # @return [Object]
     def request(method, params, timeout: 30)
-      id = request_id
-      @requests[id] = nil
-
       if !wait_until("websocket client is connected", timeout: timeout, threshold: 10.0, interval: 0.1) { connected? }
         raise TimeoutError.new(500, 'WebsocketClient is not connected')
       end
 
+      id = request_id
+      cond = Celluloid::Condition.new
+      @requests[id] = cond
+
+      start = Time.now
       websocket_client.send_request(id, method, params)
 
-      if !wait_until("request #{method} has response wth id=#{id}", timeout: timeout, interval: 0.01) { @requests[id] }
+      begin
+        cond.wait(timeout)
+      rescue Celluloid::ConditionError
+        warn "request timeout after waiting #{timeout} seconds"
```

Comment:
IOW that would be a bug from #2560, but I'm not sure exactly how the warning from the rescue shows up? If that looks reasonable, then I don't see why we need this warning... maybe improve the `raise TimeoutError` to include the timeout seconds instead?

BTW: the actual wait time can be longer than the timeout, it would be preferrable to log/raise with the `Time.now - start` instead.